src/main/java/com/jfrog/ide/common/tree/VulnerabilityOrViolation.java

```diff
@@ -0,0 +1,19 @@
+package com.jfrog.ide.common.tree;
+
+import javax.swing.tree.DefaultMutableTreeNode;
+
+public abstract class VulnerabilityOrViolation extends DefaultMutableTreeNode implements SubtitledTreeNode {
+    public abstract Severity getSeverity();
+
+    public String getIcon() {
+        return getSeverity().getIconName();
+    }
+
+    public Artifact getParentArtifact() {
+        Object parent = getParent();
```

Comment:
Is it a bad practice to use "Object" as a type
```suggestion
        TreeNode parent = getParent();
```